sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-response.integration.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from "@nestjs/common";
+import { ConfigService } from "@sims/utilities/config";
+import { SFTPIntegrationBase, SshService } from "@sims/integrations/services";
+import { ECEResponseFileDetail } from "./ece-files/ece-response-file-detail";
+import { ECEResponseFileHeader } from "./ece-files/ece-response-file-header";
+import { RecordTypeCodes } from "./models/ece-integration.model";
+import { ECEResponseFileFooter } from "./ece-files/ece-response-file-footer";
+
+@Injectable()
+export class ECEResponseIntegrationService extends SFTPIntegrationBase<
+  ECEResponseFileDetail[]
+> {
+  constructor(config: ConfigService, sshService: SshService) {
+    super(config.zoneBSFTP, sshService);
+  }
+
+  /**
+   * Downloads the file specified on {@link remoteFilePath} from the
+   * Institution response integration folder on the SFTP.
+   * @param remoteFilePath full remote file path with file name.
+   * @returns parsed records from the file.
+   */
+  async downloadResponseFile(
+    remoteFilePath: string,
+  ): Promise<ECEResponseFileDetail[]> {
+    const fileLines = await this.downloadResponseFileLines(remoteFilePath);
+    // Read the first line to check if the header record type is the expected one.
+    const header = new ECEResponseFileHeader(fileLines.shift()); // Read and remove header.
+    if (header.recordType !== RecordTypeCodes.ECEHeader) {
+      const error = `The ECE response file ${remoteFilePath} has an invalid record type on header: ${header.recordType}`;
+      this.logger.error(error);
+      // If the header is not the expected one, throw an error.
+      throw new Error(error);
+    }
+    // Read the last line to check if the footer record type is the expected one.
+    const footer = new ECEResponseFileFooter(fileLines.pop());
+    if (footer.recordType !== RecordTypeCodes.ECETrailer) {
+      const error = `The ECE response file ${remoteFilePath} has an invalid record type on footer: ${footer.recordType}`;
+      this.logger.error(error);
+      // If the footer is not the expected one.
+      throw new Error(error);
+    }
+    // The file is expected to have at least one detail record.
+    if (!fileLines.length) {
```

Comment:
@dheepak-aot just wanted to double check, In one of the grooming. I remember we were discussing 2 different types of details record errors right  1.  file lines with only header and footer no details record.  2. file lines with header, footer, and details with invalid details record identifier. Is that part of this story?

Reply:
Yes, we also discussed files with header and footer but no detail records at all.